### ISSUE TITLE
[BUG]: GitHubRelease@1 does not output error message when failing to upload artifacts to release

### ISSUE THREAD

**Issue by bluecmd** [microsoft/azure-pipelines-tasks#19599]
### New issue checklist

- [X] I searched for [existing GitHub issues](https://github.com/microsoft/azure-pipelines-tasks/issues)
- [X] I read [pipeline troubleshooting guide](https://docs.microsoft.com/vsts/build-release/actions/troubleshooting)
- [X] I checked [how to collect logs](https://learn.microsoft.com/azure/devops/pipelines/troubleshooting/review-logs?view=azure-devops)

### Task name

GitHubRelease

### Task version

1.227.0

### Issue Description

When creating a new release and uploading artifacts, if the upload fails for whatever reason the output is not descriptive.

```
[..]
2024-03-01T20:00:26.4346923Z Uploaded file successfully: '/agent/_work/1/docker-broadcom-lldp-dbg.tar.gz'
2024-03-01T20:00:26.4352569Z Uploading file: '/agent/_work/1/docker-broadcom-macsec-dbg.tar.gz'.
2024-03-01T20:05:01.2223045Z ##[error]An unexpected error occurred while uploading the file: %s
2024-03-01T20:05:01.2244868Z ##[warning]Can't find loc string for key: CreateReleaseFailed
2024-03-01T20:05:01.2245539Z CreateReleaseFailed
2024-03-01T20:05:01.2246316Z Deleting the release for tag: build-202211-455
2024-03-01T20:05:01.7539361Z Release deleted successfully.
2024-03-01T20:05:01.7540992Z ##[error]Error
2024-03-01T20:05:01.8044627Z ##[section]Finishing: Make Github release with artifacts
```

This seems to me that there is a real error returned by Github somewhere that is not being outputted, instead the message `##[error]Error` is shown. Also, the stray `%s` seems also like a bug.

### Environment type (Please select at least one enviroment where you face this issue)

- [X] Self-Hosted
- [ ] Microsoft Hosted
- [ ] VMSS Pool
- [ ] Container

### Azure DevOps Server type

dev.azure.com (formerly visualstudio.com)

### Operation system

Ubuntu 20.04

### Relevant log output

```shell
2024-03-01T19:49:49.4993612Z ##[section]Starting: Make Github release with artifacts
2024-03-01T19:49:49.5019497Z ==============================================================================
2024-03-01T19:49:49.5019710Z Task         : GitHub Release
2024-03-01T19:49:49.5019804Z Description  : Create, edit, or delete a GitHub release
2024-03-01T19:49:49.5019938Z Version      : 1.227.0
2024-03-01T19:49:49.5020042Z Author       : Microsoft Corporation
2024-03-01T19:49:49.5020139Z Help         : https://aka.ms/AA5vv5o
2024-03-01T19:49:49.5020229Z ==============================================================================
2024-03-01T19:49:49.7974981Z Computing changes made in this release...
2024-03-01T19:49:50.1128520Z Fetching the latest non-draft release...
2024-03-01T19:49:51.5207496Z Found the latest non-draft release
2024-03-01T19:49:52.6168676Z Fetching the list of commits since the last published release...
2024-03-01T19:49:52.9510492Z Found the list of changes.
2024-03-01T19:49:52.9524758Z Changes computed successfully.
2024-03-01T19:49:52.9531555Z Creating a release for tag: build-202211-455
2024-03-01T19:49:53.5286914Z Uploading assets...
2024-03-01T19:49:53.5292214Z Searching for file(s) matching '/agent/_work/1/**/target/sonic-broadcom.*'.
2024-03-01T19:49:56.4713806Z Uploading file: '/agent/_work/1/sonic-buildimage.broadcom/target/sonic-broadcom.bin'.
2024-03-01T19:51:34.2630749Z Uploaded file successfully: '/agent/_work/1/sonic-buildimage.broadcom/target/sonic-broadcom.bin'
2024-03-01T19:51:34.2632155Z Uploading file: '/agent/_work/1/sonic-buildimage.broadcom/target/sonic-broadcom.bin.log'.
2024-03-01T19:51:40.4635848Z Uploaded file successfully: '/agent/_work/1/sonic-buildimage.broadcom/target/sonic-broadcom.bin.log'
2024-03-01T19:51:40.4637013Z Searching for file(s) matching '/agent/_work/1/**/target/sonic-aboot-broadcom.*'.
2024-03-01T19:51:43.1420546Z Uploading file: '/agent/_work/1/sonic-buildimage.broadcom/target/sonic-aboot-broadcom.swi'.
2024-03-01T19:52:59.3129255Z Uploaded file successfully: '/agent/_work/1/sonic-buildimage.broadcom/target/sonic-aboot-broadcom.swi'
2024-03-01T19:52:59.3130044Z Uploading file: '/agent/_work/1/sonic-buildimage.broadcom/target/sonic-aboot-broadcom.swi.log'.
2024-03-01T19:53:00.5247120Z Uploaded file successfully: '/agent/_work/1/sonic-buildimage.broadcom/target/sonic-aboot-broadcom.swi.log'
2024-03-01T19:53:00.5248303Z Searching for file(s) matching '/agent/_work/1/**/target/sonic-mellanox.*'.
2024-03-01T19:53:03.2589615Z Uploading file: '/agent/_work/1/sonic-buildimage.mellanox/target/sonic-mellanox.bin'.
2024-03-01T19:54:48.3313746Z Uploaded file successfully: '/agent/_work/1/sonic-buildimage.mellanox/target/sonic-mellanox.bin'
2024-03-01T19:54:48.3328359Z Uploading file: '/agent/_work/1/sonic-buildimage.mellanox/target/sonic-mellanox.bin.log'.
2024-03-01T19:54:49.0374129Z Uploaded file successfully: '/agent/_work/1/sonic-buildimage.mellanox/target/sonic-mellanox.bin.log'
2024-03-01T19:54:49.0375138Z Searching for file(s) matching '/agent/_work/1/**/target/sonic-innovium.*'.
2024-03-01T19:54:51.9399136Z Uploading file: '/agent/_work/1/sonic-buildimage.innovium/target/sonic-innovium.bin'.
2024-03-01T19:56:52.5892113Z Uploaded file successfully: '/agent/_work/1/sonic-buildimage.innovium/target/sonic-innovium.bin'
2024-03-01T19:56:52.5893156Z Uploading file: '/agent/_work/1/sonic-buildimage.innovium/target/sonic-innovium.bin.log'.
2024-03-01T19:56:53.7225477Z Uploaded file successfully: '/agent/_work/1/sonic-buildimage.innovium/target/sonic-innovium.bin.log'
2024-03-01T19:56:53.7226092Z Searching for file(s) matching '/agent/_work/1/docker-*-dbg.tar.gz'.
2024-03-01T19:56:53.7283014Z Uploading file: '/agent/_work/1/docker-broadcom-database-dbg.tar.gz'.
2024-03-01T19:57:37.3789106Z Uploaded file successfully: '/agent/_work/1/docker-broadcom-database-dbg.tar.gz'
2024-03-01T19:57:37.3789650Z Uploading file: '/agent/_work/1/docker-broadcom-dhcp-relay-dbg.tar.gz'.
2024-03-01T19:58:07.0741681Z Uploaded file successfully: '/agent/_work/1/docker-broadcom-dhcp-relay-dbg.tar.gz'
2024-03-01T19:58:07.0744588Z Uploading file: '/agent/_work/1/docker-broadcom-eventd-dbg.tar.gz'.
2024-03-01T19:58:27.9972881Z Uploaded file successfully: '/agent/_work/1/docker-broadcom-eventd-dbg.tar.gz'
2024-03-01T19:58:27.9974171Z Uploading file: '/agent/_work/1/docker-broadcom-fpm-frr-dbg.tar.gz'.
2024-03-01T19:59:02.8689296Z Uploaded file successfully: '/agent/_work/1/docker-broadcom-fpm-frr-dbg.tar.gz'
2024-03-01T19:59:02.8691725Z Uploading file: '/agent/_work/1/docker-broadcom-gbsyncd-broncos-dbg.tar.gz'.
2024-03-01T19:59:30.5153319Z Uploaded file successfully: '/agent/_work/1/docker-broadcom-gbsyncd-broncos-dbg.tar.gz'
2024-03-01T19:59:30.5153841Z Uploading file: '/agent/_work/1/docker-broadcom-gbsyncd-credo-dbg.tar.gz'.
2024-03-01T19:59:57.9558843Z Uploaded file successfully: '/agent/_work/1/docker-broadcom-gbsyncd-credo-dbg.tar.gz'
2024-03-01T19:59:57.9651798Z Uploading file: '/agent/_work/1/docker-broadcom-lldp-dbg.tar.gz'.
2024-03-01T20:00:26.4346923Z Uploaded file successfully: '/agent/_work/1/docker-broadcom-lldp-dbg.tar.gz'
2024-03-01T20:00:26.4352569Z Uploading file: '/agent/_work/1/docker-broadcom-macsec-dbg.tar.gz'.
2024-03-01T20:05:01.2223045Z ##[error]An unexpected error occurred while uploading the file: %s
2024-03-01T20:05:01.2244868Z ##[warning]Can't find loc string for key: CreateReleaseFailed
2024-03-01T20:05:01.2245539Z CreateReleaseFailed
2024-03-01T20:05:01.2246316Z Deleting the release for tag: build-202211-455
2024-03-01T20:05:01.7539361Z Release deleted successfully.
2024-03-01T20:05:01.7540992Z ##[error]Error
2024-03-01T20:05:01.8044627Z ##[section]Finishing: Make Github release with artifacts
```

### Full task logs with system.debug enabled

Not available.

### Repro steps

```yml
# This is hard to repro as it requires the Github upload to fail.
# Possibly it could be attempted with an artifact that is larger than 2 GiB, which
# I believe is the largest Github size allowed.
#
# That said, here is the configuration used:

    - task: GitHubRelease@1
      displayName: Make Github release with artifacts
      inputs:
        gitHubConnection: github
        repositoryName: kamelnetworks/sonic-buildimage
        action: create
        tagSource: userSpecifiedTag
        tag: 'build-$(Build.SourceBranchName)-$(Build.BuildId)'
        title: '$(Build.SourceBranchName)-v$(Build.BuildId)'
        isPreRelease: true
        addChangeLog: true
        changeLogCompareToRelease: lastNonDraftRelease
        releaseNotesSource: inline
        releaseNotesInline: 'Build version `SONiC.$(Build.SourceBranchName)-kamel.$(Build.BuildId)-$(Build.SourceVersion)`'
        assets: |
          $(Pipeline.Workspace)/**/target/sonic-broadcom.*
          $(Pipeline.Workspace)/**/target/sonic-aboot-broadcom.*
          $(Pipeline.Workspace)/**/target/sonic-mellanox.*
          $(Pipeline.Workspace)/**/target/sonic-innovium.*
          $(Pipeline.Workspace)/docker-*-dbg.tar.gz
```

**Comment by bluecmd** [comment]
This is still an issue as far as I know

**Comment by maxnewbould-asterinsights** [comment]
Can confirm still an issue

**Comment by abhishekkumams** [comment]
This issue is still hapening.

**Comment by AvaSharkawy** [comment]
We encountered the same logging problem with `GitHubRelease@1` version **1.280.0**
on a Microsoft-hosted **windows-2022** agent using **Node 24**. Draft creation and
some uploads succeeded, then an upload stalled for about four minutes and failed
with the literal `%s` message and an otherwise empty error. Task-level retries
failed on different small files and deleted the incomplete draft each time.

We worked around it by replacing the upload operation with a small REST-based
task using the **same GitHub service connection and unchanged artifacts**:

1. Create/resume a matching draft instead of deleting the whole release on failure.
2. Upload files individually, reopening the file stream on every retry, with
   bounded timeouts and backoff for transient errors.
3. Log HTTP status, the response message and `x-github-request-id`, without headers
   containing credentials.
4. Before retrying, inspect existing assets. Keep completed assets only after
   checking their size and SHA-256. Remove failed-upload `starter` placeholders.
5. Verify the complete asset inventory before publishing. Ours intentionally
   remains a draft for testing.

The additional diagnostics exposed **HTTP 502 / "Error uploading"**, a timeout,
and a cleanup DELETE returning **500 followed by 404**. The DELETE had completed
despite its failed response, so the subsequent 404 was treated as success.
One `starter` asset reported the intended full file size, not zero, and had no
digest. Checking `size == 0` alone would miss that placeholder.

GitHub documents this [502/starter-asset behavior](https://docs.github.com/en/rest/releases/assets#upload-a-release-asset).
After these changes, **all 23 assets uploaded successfully and their remote
SHA-256 values matched**, including installers up to approximately 330 MiB.
A subsequent pipeline rerun also succeeded without re-uploading anything; all
23 asset IDs, sizes and digests remained unchanged.

There are also two relevant observations in the task source:

- [`Action.ts`](https://github.com/microsoft/azure-pipelines-tasks/blob/master/Tasks/GitHubReleaseV1/operations/Action.ts)
  calls `tl.loc("UploadAssetError")` without the asset argument, leaving `%s`
  unexpanded. It then assumes `response.body.message` exists, which can hide
  useful details for a non-JSON response.
- [`Release.ts`](https://github.com/microsoft/azure-pipelines-tasks/blob/master/Tasks/GitHubReleaseV1/operations/Release.ts)
  creates a file stream once, while
  [`webClient.ts`](https://github.com/microsoft/azure-pipelines-tasks/blob/master/Tasks/GitHubReleaseV1/operations/webClient.ts)
  retries the same request body. A consumed stream should be recreated before
  retrying an upload.

This is a confirmed workaround for our case, not proof that every occurrence of
this message has the same cause. The original task logs did not expose the HTTP
response; the 502 above was observed during the replacement uploader's diagnostic
run. Improving the built-in task's error reporting would make that distinction
much easier to establish.
